virtcontainers/container.go

```diff
@@ -551,7 +558,18 @@ func (c *Container) delete() error {
 		return err
 	}
 
-	return c.sandbox.storage.deleteContainerResources(c.sandboxID, c.id, nil)
+	if err := c.sandbox.storage.deleteContainerResources(c.sandboxID, c.id, nil); err != nil {
+		return err
+	}
+
+	// Update containers map if stateful flag.
+	if !c.sandbox.config.Stateless {
+		if err := delFromContainersMap(c.id, c.sandbox.id); err != nil {
```

Comment:
The ordering looks wrong here as we seem to have this:

- `create()`
  - create container on-disk state.
  - add container to map.

- `delete()`
  - remove container on-disk state.
  - remove container from map.

Whereas I'd expect:

- `delete()`
  - remove container from map.
  - remove container on-disk state.